File: testdata/fixtures/ocaml/CWE-295.ml
(* CWE-295: Improper Certificate Validation – disabling verification.
 * Ssl.set_verify with Verify_none disables TLS verification, allowing MITM.
 *)

let vuln ctx =
  (* ruleid: ocamllint-tls *)
  let _ = Ssl.set_verify ctx Verify_none in
  (* ruleid: ocamllint-tls *)
  let _ = Ssl.set_verify ctx verify_none in
  ()

let safe ctx =
  (* ok: ocamllint-tls *)
  let _ = Ssl.set_verify ctx Verify_peer in
  (* ok: ocamllint-tls *)
  let _ = Ssl.set_verify ctx Verify_fail_if_no_peer_cert in
  ()
let vuln2 ctx =
  (* ruleid: ocamllint-tls *)
  let _ = Ssl.set_verify ctx Verify_none in
  (* ruleid: ocamllint-tls *)
  let _ = Ssl.set_verify ctx verify_none in
  ()

let safe2 ctx =
  (* ok: ocamllint-tls *)
  let _ = Ssl.set_verify ctx Verify_peer in
  ()

File: testdata/fixtures/typescript/CWE-295.ts
// CWE-295: Improper Certificate Validation – setting rejectUnauthorized:false
// disables TLS certificate verification, allowing MITM attacks.

import https from "https";

function vuln() {
  // ruleid: typescript-tls-verification-disabled
  const opts = { rejectUnauthorized: false };
  https.request("https://example.com", opts);
  // ruleid: typescript-tls-verification-disabled
  let opts2 = { rejectUnauthorized: false };
  const _ = opts2;
  // ruleid: typescript-tls-verification-disabled
  var opts3 = { rejectUnauthorized: false };
  const __ = opts3;
}

function safe() {
  // ok: typescript-tls-verification-disabled
  const opts = { rejectUnauthorized: true };
  // ok: typescript-tls-verification-disabled
  let opts2 = { secure: true };
  // ok: typescript-tls-verification-disabled
  const x = "rejectUnauthorized: false";
  // ok: typescript-tls-verification-disabled
  const y = { rejectUnauthorized: true, ca: "cert" };
}
